grr/server/grr_response_server/gui/ui/components/online_chip/online_chip.ts

```diff
@@ -0,0 +1,43 @@
+import {Component, Input, ChangeDetectionStrategy, OnChanges, SimpleChanges} from '@angular/core';
+import {DateTime, Duration} from 'luxon';
+import {interval, Subject, merge} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+/**
+ * Component displaying the status of a Client in a material chip.
+ */
+@Component({
+  selector: 'online-chip',
+  templateUrl: './online_chip.ng.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OnlineChip implements OnChanges {
+  private static readonly STATUS_OFFLINE = 'offline';
+  private static readonly STATUS_ONLINE = 'online';
+  private static readonly ONLINE_TRESHOLD = Duration.fromMillis(1000 * 60 * 15); // 15 minutes
```

Comment:
Use `Duration.fromObject({minutes: 15});` instead - then no math and comment is needed.

Reply:
That's a lot cleaner, thanks!